doc/source/ray-core/patterns/tree-of-tasks.rst

```diff
@@ -0,0 +1,32 @@
+.. _task-pattern-tree-of-tasks:
+
+Pattern: Using a tree of tasks to achieve nested parallelism
+============================================================
+
+In this pattern, a remote function can dynamically call other remote functions (including itself) for parallelism.
```

Comment:
I think this text could use a lot more context and explanation of what things to be aware of.

- When should you want to use a tree vs a an iterative pattern?
- What are the implications of deeply nested trees?
- When can you expect the distributed version to be faster (right now it just says "note that..." but doesn't explain why this behavior happens)?